index.rst

```diff
@@ -2,28 +2,137 @@
+
+Autonomous Response With the Scheduler
+--------------------------------------
+
+In principle, in order to autonomously respond to ToOs the Scheduler CSC needs first to be configured with a Feature Based Scheduler setup that is capable or responding to inputs from a ToO alert system.
+
+The Feature Based Scheduler already has a `ToO_survey`_, that is currently successfully used for simulation purposes and that could also be used for operations.
+
+.. _ToO_survey: https://rubin-sim.lsst.io/api/rubin_sim.scheduler.surveys.ToO_survey.html#rubin_sim.scheduler.surveys.ToO_survey
+
+With that in place, the only additional element required by the Scheduler is a source of ToOs events.
+
+ToO Events as Telemetry
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently the Scheduler CSC colects telemetry from the EFD and hands them over to the ``Driver``, which is in charge of formatting it in a way the scheduling algorithm understands.
+By using a general purpose interface for colecting and passing telemetry from the EFD to the scheduling algorithm, it allows us to easily add new data sources as long as the data is in the EFD.
+
+It is natural then to expect that, any new information needed by the Scheduler, should also be in the EFD.
+
+By writting ToO events to the EFD the Scheduler CSC will naturally be able to access them.
+Next, all that is required is for the ``Driver`` be able to parse the information in a way the scheduling algorithm understands.
+
+This design also has the advantage that it allows us to control which ToO events the Scheduler should consider by filtering which events are written to the EFD, or adding metadata to indicate whether an event was certified or not.
+
+The diagram below summarizes the design of the system.
+
+.. figure:: /_static/ToO_diagram.png
+   :name: fig-too-diagram
+   :target: ../_images/ToO_diagram.png
+   :alt: ToO automatic response system design.
+
+   Design of the system to autonomously handle ToO with the Scheduler CSC.
+
+The service in charge of reading/receiving ToO alerts from external sources is the ``ToO Alert Producer``, shown in the :ref:`diagram above <fig-too-diagram>`.
+This service will also be in charge of certifying/validating which ToO events the Scheduler will consider.
+
+Furthermore, the ``ToO Alert Producer`` can be designed in such a way to support multiple ``ToO Alert Stream`` sources.
+
+Manual Response With the ScriptQueue
+------------------------------------
+
+As mentioned above, in order for the Scheduler CSC to be able to respond to a certain type of ToO, it needs to be pre-configured with a setup capable of processing the incoming data stream and acting accordingly.
+This process should work on those cases where we know what to expect and are capable of planning ahead of time (e.g. Gravitational-wave events that got pre-approved by the project).
```

Comment:
"gravitational-wave events .." --> 
(e.g. Gravitational-wave followup programs which are pre-approved by the project and have a general pre-planned format within the Feature Based Scheduler).